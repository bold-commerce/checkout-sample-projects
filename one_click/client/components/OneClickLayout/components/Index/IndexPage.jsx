import React, { useCallback, useContext, useEffect, useState }  from 'react';
import Card from '../Card';
import classNames from 'classnames';
import { Price } from '@boldcommerce/stacks-ui';
import { LineItems } from '../LineItems';
import { useCheckoutStore, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { LoadingState } from '../LoadingState';
import { Header } from '../Header';
import { AppContext } from '../../context/AppContext';
import { PaymentMethod } from '../Payment';
import { CheckoutButton } from '../CheckoutButton';
import { useAnalytics } from '../../../../hooks';

const IndexPage = ({ onSectionChange, show }, ref) => {
  const { state } = useCheckoutStore();
  const { order_total, customer, addresses, shipping } = state.applicationState;
  const shippingAddressArray = [ addresses.shipping.address_line_1, addresses.shipping.address_line_2, addresses.shipping.city, addresses.shipping.province, addresses.shipping.postal_code, addresses.shipping.country ]
  const billingAddressArray = [ addresses.billing.address_line_1, addresses.billing.address_line_2, addresses.billing.city, addresses.billing.province, addresses.billing.postal_code, addresses.billing.country ];
  const shippingCombined = shippingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const billingCombined = billingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const { submitShippingAddress } = useShippingAddress();
  const [loading, setLoading] = useState(false);
  const { websiteName } = useContext(AppContext);
  const trackEvent = useAnalytics();

  const setDefaultAddress = useCallback(async () => {
    setLoading(true);
    try {
      await submitShippingAddress(customer.saved_addresses[0]);
    } catch(e) {
      setLoading(false);
    }
    setLoading(false);
  }, [customer.saved_addresses]);

  useEffect(() => {
    //if customer hasn't set a shipping address yet and they have a saved shipping address, set the shipping address to the first one. 
    if(addresses.shipping.length == 0 && customer.saved_addresses.length > 0){
      setDefaultAddress();
    }

    trackEvent('landing_page');
  }, []);

  return (
    <div ref={ref} className={classNames('Sidebar IndexPage', show ? 'Sidebar--Show' : 'IndexPage--Hide')}>
      <Header title={websiteName}/>
      <LineItems />
      <Card
        title={"Summary"}
        handleClick={() => onSectionChange('summary')}
        component={"/summary"}
        overview={<Price amount={order_total} />}
      />
      { customer?.email_address && (
        <Card
          description={customer.email_address}
          action={{label: "Not you?"}}
        />
      )}
      <Card
        title={"Shipping"}
        type={"shippingCard"}
        handleClick={() => onSectionChange('shipping')}
        description={loading ? '' : addresses.shipping.first_name ? `${addresses.shipping.first_name} ${addresses.shipping.last_name}` : 'No shipping address selected'}
      >
      { (loading && <LoadingState/>) || 
      ( 
        shipping.selected_shipping &&
        <>
          <div>{shippingCombined}</div>
          <div className="card-shipping-content">{shipping.selected_shipping.description} - <Price amount={shipping.selected_shipping.amount} /></div>
        </>
      )}
      </Card>
      <Card
        title={"Payment"}
        type={"paymentCard"}
      >
      {
        state.orderInfo.billingSameAsShipping ? 
        <div>Billing address same as shipping address</div> :
        <div>{billingCombined}</div>
      }
      </Card>
      <PaymentMethod />
      <CheckoutButton className="CheckoutButton CheckoutButton__Mobile" />
    </div>
  )
};

const IndexPageForwardedRef = React.forwardRef(IndexPage);

export default IndexPageForwardedRef;
