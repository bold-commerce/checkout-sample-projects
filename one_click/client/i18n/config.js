import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;