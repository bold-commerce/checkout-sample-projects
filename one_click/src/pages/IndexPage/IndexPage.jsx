import React, { useCallback, useContext, useEffect, useState }  from 'react';
import Card from '../../components/Card';
import classNames from 'classnames';
import { Price } from '@boldcommerce/stacks-ui';
import { LineItems } from '../../components/LineItems';
import { useCheckoutStore, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { LoadingState } from '../../components/LoadingState';
import { Header } from '../../components/Header';
import { AppContext } from '../../context/AppContext';
import { PaymentMethod } from '../../components/Payment';
import { CheckoutButton } from '../../components/CheckoutButton';
import { useAnalytics } from '../../hooks';
import { useTranslation } from 'react-i18next';
import { SummaryCondensed } from '../../components/Summary';

const IndexPage = ({ onSectionChange, show }, ref) => {
  const { state } = useCheckoutStore();
  const { order_total, customer, addresses, shipping } = state.applicationState;
  const shippingAddressArray = [ addresses.shipping.address_line_1, addresses.shipping.address_line_2, addresses.shipping.city, addresses.shipping.province, addresses.shipping.postal_code, addresses.shipping.country ]
  const billingAddressArray = [ addresses.billing.address_line_1, addresses.billing.address_line_2, addresses.billing.city, addresses.billing.province, addresses.billing.postal_code, addresses.billing.country ];
  const shippingCombined = shippingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const billingCombined = billingAddressArray.filter((addressString) => addressString !== '').join(', ');
  const { submitShippingAddress } = useShippingAddress();
  const [loading, setLoading] = useState(false);
  const { websiteName } = useContext(AppContext);
  const trackEvent = useAnalytics();
  const { t } = useTranslation();

  const setDefaultAddress = useCallback(async () => {
    setLoading(true);
    try {
      await submitShippingAddress(customer.saved_addresses[0]);
    } catch(e) {
      setLoading(false);
    }
    setLoading(false);
  }, [customer.saved_addresses]);

  useEffect(() => {
    //if customer hasn't set a shipping address yet and they have a saved shipping address, set the shipping address to the first one. 
    if(addresses.shipping.length == 0 && customer.saved_addresses.length > 0){
      setDefaultAddress();
    }

    trackEvent('landing_page');
  }, []);

  return (
    <div ref={ref} className={classNames('Sidebar IndexPage', show ? 'Sidebar--Show' : 'IndexPage--Hide')}>
      <Header title={websiteName}/>
      <LineItems />
      <SummaryCondensed handleClick={() => onSectionChange('summary')}/>
      { customer?.email_address && (
        <Card
          description={customer.email_address}
          action={{
            label: t('customer.not_you'),
            onClick: () => { window.location = process.env.LOGOUT_URL }
          }}
        />
      )}
      <Card
        title={t('shipping.title')}
        type={"shippingCard"}
        handleClick={() => onSectionChange('shipping')}
        description={loading ? '' : addresses.shipping.first_name ? `${addresses.shipping.first_name} ${addresses.shipping.last_name}` : t('shipping.empty')}
      >
      { (loading && <LoadingState/>) || 
      ( 
        shipping.selected_shipping &&
        <>
          <div>{shippingCombined}</div>
          <div className="card-shipping-content">{shipping.selected_shipping.description} - <Price amount={shipping.selected_shipping.amount} moneyFormatString={t('currency_format')}/></div>
        </>
      )}
      </Card>
      <Card
        title={t('payment.title')}
        type={"paymentCard"}
      >
      {
        state.orderInfo.billingSameAsShipping ? 
        <div>{t('payment.billing_same_as_shipping')}</div> :
        <div>{billingCombined}</div>
      }
      </Card>
      <PaymentMethod />
      <CheckoutButton className="CheckoutButton CheckoutButton__Mobile" />
    </div>
  )
};

const IndexPageForwardedRef = React.forwardRef(IndexPage);

export default IndexPageForwardedRef;
