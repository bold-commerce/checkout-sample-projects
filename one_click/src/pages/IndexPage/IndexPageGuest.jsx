import React, { useContext, useState } from "react";
import { Button } from "@boldcommerce/stacks-ui";
import { useShippingAddress, useCountryInfo, useCheckoutStore, useCustomer } from "@boldcommerce/checkout-react-components";
import { ShippingLines } from "../../components/ShippingLines";
import { AppContext } from "../../context/AppContext";
import { Address } from "../../components/Address";
import { Header } from "../../components/Header";
import { Link } from "react-router-dom";
import classNames from "classnames";
import { CustomerInfo } from "../../components/CustomerInfo";
import { SummaryCondensed } from "../../components/Summary";
import { useTranslation } from 'react-i18next';

const IndexPageGuest = ({ onSectionChange, show }, ref) => {
  const { websiteName } = useContext(AppContext);
  const { state } = useCheckoutStore();
  const { data: shippingAddress, errors: shippingErrors, submitShippingAddress } = useShippingAddress();
  const [ address, setAddress ] = useState(shippingAddress);
  const { data: countryInfo } = useCountryInfo(address);
  const { data: customer, errors: customerErrors } = useCustomer();
  const { t } = useTranslation();

  const requiredAddressFields = ['first_name','last_name','address_line_1','city'];
  let provincePlaceholder = countryInfo.provinceLabel;

  return (
    <div ref={ref} className={classNames('Sidebar IndexGuest', show ? 'Sidebar--Show' : 'IndexPage--Hide')}>
      <Header title={websiteName} />
      <SummaryCondensed onSectionChange={() => onSectionChange('summary')}/>
      <CustomerInfo />
      <div className="IndexGuest__ShippingAddress">
        <h2 className="IndexGuest__Title">{t('shipping.address')}</h2>
        <Address
          address={address}
          onChange={(d) => setAddress((prevAddress) => ({
            ...prevAddress,
            ...d,
          }))}
          errors={shippingErrors}
          countries={countryInfo.countries}
          provinces={countryInfo.provinces}
          showPostalCode={countryInfo.showPostalCode}
          showProvince={countryInfo.showProvince}
          submit={() => submitShippingAddress(address)}
          provinceLabel={provincePlaceholder}
          requiredAddressFields={requiredAddressFields}
        />
      </div>
      <div className="IndexGuest__ShippingMethod">
        <ShippingLines activePage={show} disabled={Boolean(shippingErrors)} />
      </div>
      <div className="IndexGuest__Footer">
        <Button
          className="IndexGuest__Billing-btn"
          onClick={() => onSectionChange('billing')}
          disabled={!(customer.email_address && state.applicationState.shipping.selected_shipping && !shippingErrors && !customerErrors)}
          primary
        >
          {t('continue_to_billing')}
        </Button>
        <div className="IndexGuest__Footer--login">
          <Link to="#">{t('back_to_login')}</Link>
        </div>          
        <div className="IndexGuest__Rights">{`${t('all_rights_reserved')} ${websiteName}`}</div>
      </div>
    </div>
  )
}

const IndexPageGuestForwardedRef = React.forwardRef(IndexPageGuest);

export default IndexPageGuestForwardedRef;