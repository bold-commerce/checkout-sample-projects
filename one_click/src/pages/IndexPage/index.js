export { default as IndexPage } from "./IndexPage";
export { default as IndexPageGuest } from "./IndexPageGuest";