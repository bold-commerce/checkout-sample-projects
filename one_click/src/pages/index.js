export * from './Confirmation';
export * from './IndexPage';
export * from './Processing';
export * from './Inventory';