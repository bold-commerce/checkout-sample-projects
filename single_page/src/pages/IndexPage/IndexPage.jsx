import React, { useCallback, useEffect, useState } from 'react';
import { useShippingAddress, useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { useAnalytics, useErrorLogging } from '../../hooks';
import { OrderErrors, Customer, ShippingAddress, BillingAddress, ShippingLines, PaymentMethod, CheckoutButton, OrderSummary } from '../../components';
import classNames from 'classnames';

const IndexPage = () => {
  const { state } = useCheckoutStore();
  const { submitShippingAddress } = useShippingAddress();
  const trackEvent = useAnalytics();
  const logError = useErrorLogging();
  const [loading, setLoading] = useState(false);
  const [summaryOpen, setSummaryOpen] = useState(false);

  const setDefaultAddress = useCallback(async () => {
    setLoading(true);
    try {
      await submitShippingAddress(state.applicationState.customer.saved_addresses[0]);
    } catch(e) {
      logError('shipping_address', e);
      setLoading(false);
    }
    setLoading(false);
  }, [state]);

  useEffect(() => {
    // If customer is logged in with saved addresses, default the shipping address to the first one.
    if (state.applicationState.customer.saved_addresses.length > 0) {
      setDefaultAddress();
    }

    trackEvent('landing_page');
  }, []);

  return (
    <>
      <div className="Checkout__Main" role="main">
        <OrderErrors />
        <Customer />
        <ShippingAddress applicationLoading={loading} />
        <BillingAddress applicationLoading={loading} />
        <ShippingLines applicationLoading={loading} />
        <PaymentMethod applicationLoading={loading} />
        <CheckoutButton />
      </div>
      <div
        className={classNames('Checkout__Sidebar', summaryOpen ? 'Checkout__Sidebar--Open' : 'Checkout__Sidebar--Closed' )}
        role="complementary"
      >
        <OrderSummary readOnly={false} summaryOpen={summaryOpen} onCollapse={() => setSummaryOpen((prevState) => !prevState)} />
      </div>
    </>
  );
};

export default IndexPage;
