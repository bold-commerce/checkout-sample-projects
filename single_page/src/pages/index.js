export * from './ConfirmationPage';
export * from './IndexPage';
export * from './ProcessingPage';
