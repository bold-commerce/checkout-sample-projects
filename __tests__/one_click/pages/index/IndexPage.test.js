import React from 'react';
import { render } from '@testing-library/react';
import { IndexPage } from '../../../../one_click/src/pages/IndexPage';
import {
    exampleUseBillingSameAsShipping as MOCKbillingSameAsShipping,
    exampleUseShippingAddress as MOCKshippingAddress,
    exampleUseBillingAddress as MOCKbillingAddress,
    exampleUseCheckoutStore as MOCKcheckoutStore,
    exampleUsePaymentIframe as MOCKpaymentIframe,
    exampleUseShippingLines as MOCKshippingLines,
    exampleUseCountryInfo as MOCKcountryInfo,
    exampleUseLineItems as MOCKlineItems,
    exampleUseDiscount as MOCKdiscount,
} from '../../../utils/hookHelpers';
import '../../../../one_click/src/i18n/config';
import { AppContext } from '../../../../one_click/src/context/AppContext';
import { MemoryRouter } from 'react-router-dom';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useBillingSameAsShipping: () => MOCKbillingSameAsShipping,
    useShippingAddress: () => MOCKshippingAddress,
    useBillingAddress: () => MOCKbillingAddress,
    useCheckoutStore: () => MOCKcheckoutStore,
    usePaymentIframe: () => MOCKpaymentIframe,
    useShippingLines: () => MOCKshippingLines,
    useCountryInfo: () => MOCKcountryInfo,
    useLineItems: () => MOCKlineItems,
    useDiscount: () => MOCKdiscount,
}));

describe('IndexPage', () => {
    test('renders IndexPage component', () => {
        const { asFragment } = render(
            <AppContext.Provider value={{ websiteName: "app.test" }}>
                <IndexPage />
            </AppContext.Provider>,
            {wrapper: MemoryRouter}
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
