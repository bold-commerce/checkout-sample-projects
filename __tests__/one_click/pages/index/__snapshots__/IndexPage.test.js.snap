// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders IndexPage component 1`] = `
<DocumentFragment>
  <div
    class="Sidebar IndexPage IndexPage--Hide"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        TestSite
      </h1>
      <button
        aria-label="Close"
        class="Header__Times"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="SummaryBlock CartItem"
    >
      <div
        class="CartItem__ImageDescriptionWrapper"
      >
        <div
          class="CartItem__ProductImageWrapper"
        >
          <div
            class="stx-image"
          >
            <div
              class="stx-image--content"
            >
              <div
                class="stx-image--holder"
              >
                <img
                  alt="Test title"
                  class="stx-image--image"
                  src=""
                  title="Test title"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="CartItem__ProductDetails"
        >
          <div
            class="stx-details"
          >
            <div
              class="stx-details__title"
            >
              Test title
            </div>
            <div
              class="stx-details__description"
            >
              <div>
                Size Large
              </div>
              <div>
                Colour Blue
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
      <div
        class="CartItem__QuantityPriceWrapper"
      >
        <div
          class="CartItem__ProductQuantityWrapper"
        >
          <div
            aria-label="product quantity"
            class="ProductQuantity"
          >
            1
          </div>
        </div>
        <div
          class="CartItem__ProductPrice"
        >
          <div
            class="stx-price stx-price--alight-center"
          >
            <p
              aria-label="Product price"
              class="stx-price__amount"
            >
              $14.00 CAD
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="SummaryBlock CartItem"
    >
      <div
        class="CartItem__ImageDescriptionWrapper"
      >
        <div
          class="CartItem__ProductImageWrapper"
        >
          <div
            class="stx-image"
          >
            <div
              class="stx-image--content"
            >
              <div
                class="stx-image--holder"
              >
                <img
                  alt="Test title 2, Revenge of the title"
                  class="stx-image--image"
                  src=""
                  title="Test title 2, Revenge of the title"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="CartItem__ProductDetails"
        >
          <div
            class="stx-details"
          >
            <div
              class="stx-details__title"
            >
              Test title 2, Revenge of the title
            </div>
            <div
              class="stx-details__description"
            >
              <div>
                Number Two
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
      <div
        class="CartItem__QuantityPriceWrapper"
      >
        <div
          class="CartItem__ProductQuantityWrapper"
        >
          <div
            aria-label="product quantity"
            class="ProductQuantity"
          >
            2
          </div>
        </div>
        <div
          class="CartItem__ProductPrice"
        >
          <div
            class="stx-price stx-price--alight-center"
          >
            <p
              aria-label="Product price"
              class="stx-price__amount"
            >
              $32.00 CAD
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="SummaryBlock CartItem"
    >
      <div
        class="CartItem__ImageDescriptionWrapper"
      >
        <div
          class="CartItem__ProductImageWrapper"
        >
          <div
            class="stx-image"
          >
            <div
              class="stx-image--content"
            >
              <div
                class="stx-image--holder"
              >
                <img
                  alt="Test title 3, the search for more money"
                  class="stx-image--image"
                  src=""
                  title="Test title 3, the search for more money"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="CartItem__ProductDetails"
        >
          <div
            class="stx-details"
          >
            <div
              class="stx-details__title"
            >
              Test title 3, the search for more money
            </div>
            <div
              class="stx-details__description"
            />
          </div>
          <div />
        </div>
      </div>
      <div
        class="CartItem__QuantityPriceWrapper"
      >
        <div
          class="CartItem__ProductQuantityWrapper"
        >
          <div
            aria-label="product quantity"
            class="ProductQuantity"
          >
            3
          </div>
        </div>
        <div
          class="CartItem__ProductPrice"
        >
          <div
            class="stx-price stx-price--alight-center"
          >
            <p
              aria-label="Product price"
              class="stx-price__amount"
            >
              $123.00 CAD
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Card"
    >
      <div
        class="Card__Title"
      >
        <button
          type="button"
        >
          Summary
        </button>
      </div>
      <div
        class="Card__Overview"
      >
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            $262.00 CAD
          </p>
        </div>
      </div>
      <div
        class="Card__Description"
      />
      <div
        class="Card__Action"
      />
      <div
        class="Card__Content"
      >
        <div />
      </div>
    </div>
    <div
      class="Card"
    >
      <div
        class="Card__Title"
      />
      <div
        class="Card__Overview"
      />
      <div
        class="Card__Description"
      >
        john.doe@email.com
      </div>
      <div
        class="Card__Action"
      >
        Not you?
      </div>
      <div
        class="Card__Content"
      >
        <div />
      </div>
    </div>
    <div
      class="Card Card--Shipping"
    >
      <div
        class="Card__Title"
      >
        <button
          type="button"
        >
          Shipping
        </button>
      </div>
      <div
        class="Card__Overview"
      />
      <div
        class="Card__Description"
      >
        John Doe
      </div>
      <div
        class="Card__Action"
      />
      <div
        class="Card__Content"
      >
        <div>
          <div>
            123 Fake Street, Winnipeg, Manitoba, R3T 1B8, Canada
          </div>
          <div
            class="card-shipping-content"
          >
            Standard - 
            <div
              class="stx-price stx-price--alight-center"
            >
              <p
                aria-label="Product price"
                class="stx-price__amount"
              >
                $10.00 CAD
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Card Card--Payment"
    >
      <div
        class="Card__Title"
      >
        <button
          type="button"
        >
          Payment
        </button>
      </div>
    </div>
    <div
      class="Payment"
    >
      <section
        class="FieldSet FieldSet--PaymentMethod"
      >
        <div
          class="FieldSet__Content"
        >
          <iframe
            data-bold-pigi-iframe="true"
            src="test.url"
            title="payments"
          />
        </div>
        <div
          class="FieldSet__Content"
        >
          <div
            class="DiscountLink"
          >
            <button
              aria-label="add discount code"
            >
              <svg
                aria-labelledby="button"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.3335 2.66667C7.3335 2.29848 7.63197 2 8.00016 2C8.36835 2 8.66683 2.29848 8.66683 2.66667V13.3333C8.66683 13.7015 8.36835 14 8.00016 14C7.63197 14 7.3335 13.7015 7.3335 13.3333V2.66667Z"
                  fill="#0A75C2"
                />
                <path
                  d="M2.66667 8.66665C2.29848 8.66665 2 8.36817 2 7.99998C2 7.63179 2.29848 7.33331 2.66667 7.33331H13.3333C13.7015 7.33331 14 7.63179 14 7.99998C14 8.36817 13.7015 8.66665 13.3333 8.66665H2.66667Z"
                  fill="#0A75C2"
                />
              </svg>
              Discount code
            </button>
          </div>
        </div>
        <div
          class="FieldSet__Content"
        >
          <section
            class="FieldSet FieldSet--BillingAddress"
          >
            <div
              class="FieldSet__Header"
            >
              <h3
                class="Billing__Title"
              >
                Billing address
              </h3>
            </div>
            <div
              class="FieldSet__Content"
            >
              <div
                class="stx-field stx-checkbox-field__field RadioButton"
              >
                <div
                  class="stx-checkbox-field__div"
                >
                  <span
                    class="stx-checkbox stx-checkbox-field__checkbox"
                  >
                    <input
                      class="stx-checkbox__input"
                      id="stxField1"
                      name="billing-address"
                      type="checkbox"
                      value="SAME_AS_SHIPPING_ADDRESS"
                    />
                    <span
                      class="stx-checkbox__mount"
                    />
                  </span>
                  <label
                    class="stx-checkbox-field__label"
                    for="stxField1"
                  >
                    Same as shipping address
                  </label>
                </div>
              </div>
              <div
                class="FieldSet--Address"
              >
                <div
                  class="FieldGroup"
                >
                  <div
                    class="stx-field stx-field--with-input Field Field--FirstName"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="given-name"
                        class="stx-input__input-element"
                        id="stxField2"
                        name="first_name"
                        placeholder="First name (optional)"
                        type="text"
                        value="John"
                      />
                    </div>
                  </div>
                  <div
                    class="stx-field stx-field--with-input Field Field--LastName"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="family-name"
                        class="stx-input__input-element"
                        id="stxField3"
                        name="last_name"
                        placeholder="Last name (optional)"
                        type="text"
                        value="Doe"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="FieldGroup"
                >
                  <div
                    class="stx-field stx-field--with-input Field Field--Company"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        class="stx-input__input-element"
                        id="stxField4"
                        name="business_name"
                        placeholder="Company (optional)"
                        type="text"
                        value="Bold Commerce"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="FieldGroup"
                >
                  <div
                    class="stx-field stx-field--with-input Field Field--Address"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="address_line1"
                        class="stx-input__input-element"
                        id="stxField5"
                        name="address_line_1"
                        placeholder="Address (optional)"
                        type="text"
                        value="123 Fake Street"
                      />
                    </div>
                  </div>
                  <div
                    class="stx-field stx-field--with-input Field Field--Address2"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="address_line2"
                        class="stx-input__input-element"
                        id="stxField6"
                        name="address_line_2"
                        placeholder="Apt, suite, etc."
                        type="text"
                        value="Unit 12"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="FieldGroup"
                >
                  <div
                    class="stx-field stx-field--with-input Field Field--City"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="address-level2"
                        class="stx-input__input-element"
                        id="stxField7"
                        name="city"
                        placeholder="City (optional)"
                        type="text"
                        value="Winnipeg"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="FieldGroup FieldGroup__Country"
                >
                  <div
                    class="stx-field stx-field--with-select SelectField Field--Country"
                  >
                    <div
                      class="stx-select stx-select--with-placeholder"
                    >
                      <select
                        autocomplete="country"
                        class="stx-select__select-element"
                        id="stxField8"
                        name="country"
                      >
                        <option
                          hidden=""
                          value="___PLACEHOLDER___"
                        >
                          Select a country
                        </option>
                        <option
                          value="CA"
                        >
                          Canada
                        </option>
                        <option
                          value="US"
                        >
                          United States
                        </option>
                      </select>
                      <span
                        class="stx-select__icon"
                      >
                        <svg
                          class="stx-select-icon__icon"
                          viewBox="0 0 24 24"
                        >
                          <path
                            class="stx-select-icon__path"
                            d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                            fill="#6B6B6B"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="stx-field stx-field--with-select SelectField Field--Province"
                  >
                    <div
                      class="stx-select stx-select--with-placeholder"
                    >
                      <select
                        autocomplete="address-level1"
                        class="stx-select__select-element"
                        id="stxField9"
                        name="province"
                      >
                        <option
                          hidden=""
                          value="___PLACEHOLDER___"
                        >
                          Select a province
                        </option>
                        <option
                          value="AB"
                        >
                          Alberta
                        </option>
                        <option
                          value="BC"
                        >
                          British Columbia
                        </option>
                        <option
                          value="MB"
                        >
                          Manitoba
                        </option>
                        <option
                          value="NB"
                        >
                          New Brunswick
                        </option>
                        <option
                          value="NL"
                        >
                          Newfoundland and Labrador
                        </option>
                        <option
                          value="NT"
                        >
                          Northwest Territories
                        </option>
                        <option
                          value="NS"
                        >
                          Nova Scotia
                        </option>
                        <option
                          value="NU"
                        >
                          Nunavut
                        </option>
                        <option
                          value="ON"
                        >
                          Ontario
                        </option>
                        <option
                          value="PE"
                        >
                          Prince Edward Island
                        </option>
                        <option
                          value="QC"
                        >
                          Quebec
                        </option>
                        <option
                          value="SK"
                        >
                          Saskatchewan
                        </option>
                        <option
                          value="YT"
                        >
                          Yukon Territory
                        </option>
                      </select>
                      <span
                        class="stx-select__icon"
                      >
                        <svg
                          class="stx-select-icon__icon"
                          viewBox="0 0 24 24"
                        >
                          <path
                            class="stx-select-icon__path"
                            d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                            fill="#6B6B6B"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                  <div
                    class="stx-field stx-field--with-input Field Field--Postal_Code"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="postal-code"
                        class="stx-input__input-element"
                        id="stxField10"
                        name="postal"
                        placeholder="Postal code"
                        type="text"
                        value="R3A1B3"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="FieldGroup"
                >
                  <div
                    class="stx-field stx-field--with-input Field Field--Phone"
                  >
                    <div
                      class="stx-input"
                    >
                      <input
                        autocomplete="tel"
                        class="stx-input__input-element"
                        id="stxField11"
                        name="phone_number"
                        placeholder="Phone (optional)"
                        type="tel"
                        value="204-123-4567"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </section>
    </div>
    <div
      class="CheckoutButton__Container"
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="stx-button CheckoutButton CheckoutButton__Mobile"
      >
        <span
          class="stx-button__text"
        >
          Complete Order
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
